viadot/sources/genesys.py

```diff
@@ -0,0 +1,526 @@
+            raise APIError("Failed to loaded all exports.")
+
+    def get_reporting_exports_data(self):
+        """
+        Function that generate list of reports metadata for further processing steps,
+        like deteling old reports by providing list of them.
+        """
+        request_json = self.load_reporting_exports()
+
+        if request_json is not None:
+            entities = request_json.get("entities")
+            assert type(entities) == list
+            if len(entities) != 0:
+                for entity in entities:
+                    tmp = [
+                        entity.get("id"),
+                        entity.get("downloadUrl"),
+                        entity.get("filter").get("queueIds")[0],
+                        entity.get("filter").get("mediaTypes")[0],
+                    ]
+                    self.report_data.append(tmp)
+            assert len(self.report_data) > 0
+        self.logger.info("Generated list of reports entities.")
+
+    def download_report(
+        self,
+        report_url: str,
+        output_file_name: str = None,
+        file_extension: str = "csv",
+        path: str = "",
+        sep: str = "\t",
+        drop_duplicates: bool = True,
+    ):
+        """Download report to excel file.
+
+        Args:
+            report_url (str): url to report, fetched from json response.
+            output_file_name (str, optional): Output file name. Defaults to None.
+            file_extension (str, optional): Output file extension. Defaults to "xls".
+            path (str, optional): Path to the generated excel file. Defaults to empty string.
+            sep (str, optional): Separator in csv file. Defaults to "\t".
+            drop_duplicates (bool, optional): Decide if drop duplicates. Defaults to True.
+        """
+        response_file = handle_api_response(
+            url=f"{report_url}", headers=self.authorization_token
+        )
+        if output_file_name is None:
+            final_file_name = f"Genesys_Queue_Metrics_Interval_Export.{file_extension}"
+        else:
+            final_file_name = f"{output_file_name}.{file_extension}"
+
+        response_data = response_file.content
+        df = pd.read_csv(StringIO(response_file.content.decode("utf-8")))
+        if drop_duplicates is True:
+            df.drop_duplicates(inplace=True, ignore_index=True)
+        df.to_csv(f"{path}{final_file_name}", index=False, sep=sep)
+
+    def download_all_reporting_exports(
+        self,
+        store_file_names: bool = True,
+    ) -> List[str]:
+        """
+        Get information form data report and download all files.
+
+        Args:
+            g_instance (Genesys): instance of Genesys source
+            ids_mapping (Dict[str, Any], optional): relationship between id and file name. Defaults to None.
+            data_report (List[str], optional): data extracted from genesys. Defaults to None.
```

Comment:
You have `g_instance` and `data_report` in docstring but not in a function declaration. You don't use these variables in the function.